src/main/java/TaskList.java

```diff
@@ -0,0 +1,186 @@
+            return false;
+        } else {
+            return Integer.parseInt(day) <= maxDay;
+        }
+    }
+
+    /**
+     * Adds a task into the task list.
+     *
+     * @param type the type of task to be added
+     * @param name the name of task to be added
+     * @param description the description of the task
+     * @throws DukeException
+     */
+    public void addTask(String type, String name, String description) throws DukeException {
+        switch (type) {
+        case "todo":
+            if (name.trim().equals("")) {
+                throw new DukeException("No task name");
+            }
+            Task newestTodo = new ToDo(name);
+            taskArrayList.add(newestTodo);
+            System.out.println(sandwich("New todo task added:\n"
+                    + newestTodo
+                    + "\nYou now have "
+                    + taskArrayList.size()
+                    + " item(s) in your task list."));
+            break;
+        case "deadline":
+            if (!isValidDate(description)) {
+                throw new DukeException("Invalid Date, please follow the format YYYY-MM-DD");
+            }
+            Task newestDeadline = new Deadline(name, description);
+            taskArrayList.add(newestDeadline);
+            System.out.println(sandwich("New deadline task added:\n"
+                    + newestDeadline
+                    + "\nYou now have "
+                    + taskArrayList.size()
+                    + " item(s) in your task list."));
+            break;
+        case "event":
+            if (!isValidDate(description)) {
+                throw new DukeException("Invalid Date, please follow the format YYYY-MM-DD");
+            }
+            Task newestEvent = new Event(name, description);
+            taskArrayList.add(newestEvent);
+            System.out.println(sandwich("New deadline task added:\n"
+                    + newestEvent
+                    + "\nYou now have "
+                    + taskArrayList.size()
+                    + " item(s) in your task list."));
+            break;
+        default:
+            throw new DukeException("Unknown task type");
+        }
+    }
+
+    public void listTasks() {
+        System.out.println(sandwich(listBeautify(taskArrayList)));
+    }
+
+    public void finishTask(int index) throws DukeException {
+        if (index > taskArrayList.size()) {
+            throw new DukeException("This task index is not in the task list!");
+        }
+        taskArrayList.get(index - 1).markAsDone();
+        System.out.println(sandwich("Congratulations! You have finished this task: "
+                + taskArrayList.get(index - 1).toString()));
+    }
+
+    public void deleteTask(int index) throws DukeException {
+        if (index > taskArrayList.size()) {
+            throw new DukeException("This task index is not in the task list!");
+        }
+        System.out.println(sandwich("Got it, I have deleted this task: "
```

Comment:
Maybe you could refactor this code and similar methods to fit in line with SLAP Coding Quality? (Single Level of Abstraction Principle)